x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -795,10 +796,15 @@ const updateExistingDataStream = async ({
 
     // if update fails, rollover data stream and bail out
   } catch (err) {
-    logger.info(`Mappings update for ${dataStreamName} failed due to ${err}`);
-    logger.info(`Triggering a rollover for ${dataStreamName}`);
-    await rolloverDataStream(dataStreamName, esClient);
-    return;
+    if (isResponseError(err) && err.statusCode === 400 && err.message.toLowerCase().includes('mapper')) {
```

Comment:
the mapper check is not good enough, there can be a lot other reasons why the mapping update can fail: https://github.com/elastic/elasticsearch/blob/21d9de04a1c00b5c8524ae0639909c134ce0c69f/server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java#L323